examples/channels-correlation/channels_correlation.bal

```diff
@@ -2,40 +2,48 @@ import ballerina/http;
 import ballerina/log;
 
 // Defines a channel with json constrained type.
-channel<json> jsonChannel;
+channel<json> jsonChannel = new;
 
-service<http:Service> channelService bind { port: 9090 } {
+service channelService on new http:Listener(9090) {
 
-    receive(endpoint caller, http:Request request) {
+    resource function receive(http:Caller caller, http:Request request) {
 
         // A key can be associated with a channel action.
         // You can use a key to correlate receivers and senders.
         string key = "123";
 
-        json result;
+        json jsonMsg;
         // Receive a message from the channel with given key.
         // Execution waits here if the message is not available.
-        result <- jsonChannel, key;
+        jsonMsg = <- jsonChannel, key;
         // Send the received message as the response.
-        var result = caller->respond(result);
+        var result = caller->respond(jsonMsg);
 
         if (result is error) {
             log:printError("Error sending response", err = result);
         }
     }
 
-    send(endpoint caller, http:Request request) {
+    resource function send(http:Caller caller, http:Request request) {
 
         // Extract message from the request.
-        json message = check request.getJsonPayload();
+        json|error message = request.getJsonPayload();
         // Define the same key as the receiver's key.
         string key = "123";
 
         // Send a message to the channel.
         // One of the receivers waiting on this key receives it.
         // If there is no receiver, the message is stored and execution continues.
         // A receiver can arrive later and fetch the message.
-        message -> jsonChannel, key;
+
+        json jsonMessage = {};
+        if (message is json) {
+            jsonMessage = message;
+        } else {
+            log:printError("Invalid message content");
```

Comment:
should we encourage to print the error details here as well ?

i.e } else if (message is error){
          log:printError("Invalid message content", err=message);
     }